modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -2383,6 +2424,35 @@ private CompletableFuture<Object> applyCmdWithExceptionHandling(Command cmd) {
         });
     }
 
+    private void applyCmdWithRetryOnSafeTimeReorderException(Command cmd, CompletableFuture<Object> resultFuture) {
+        raftClient.run(cmd).whenComplete((res, ex) -> {
+            if (ex != null) {
+                if (ex instanceof SafeTimeReorderException || ex.getCause() instanceof SafeTimeReorderException) {
+                    assert cmd instanceof SafeTimePropagatingCommand;
```

Comment:
Do you mean that applyCmdWithRetryOnSafeTimeReorderException should have SafeTimePropagatingCommand instead of Command as a parameter? I don't think it's convenient because in that case we should skip calling  applyCmdWithRetryOnSafeTimeReorderException for all commands except SafeTimePropagatingCommand and skip raftClientRun directly.